pkg/console/operator/status.go

```diff
@@ -66,20 +58,38 @@ const (
 //   RouteStatusDegraded
 //   CustomLogoInvalidDegraded
 // and uses the error as its message
-func (c *consoleOperator) HandleDegraded(operatorConfig *operatorsv1.Console, prefix string, err error) {
-	conditionType := prefix + operatorsv1.OperatorStatusTypeDegraded
-	reason := prefix + "Error"
+func (c *consoleOperator) HandleDegraded(operatorConfig *operatorsv1.Console, prefix ConditionPrefix, err error) {
+	prefixString := string(prefix)
+	conditionType := prefixString + operatorsv1.OperatorStatusTypeDegraded
+	reason := prefixString + "Error"
 	handleCondition(operatorConfig, conditionType, reason, err)
 }
 
-func (c *consoleOperator) HandleProgressing(operatorConfig *operatorsv1.Console, prefix string, err error) {
-	conditionType := prefix + operatorsv1.OperatorStatusTypeProgressing
-	handleCondition(operatorConfig, conditionType, prefix, err)
+func (c *consoleOperator) HandleProgressing(operatorConfig *operatorsv1.Console, prefix ConditionPrefix, err error) {
+	prefixString := string(prefix)
+	conditionType := prefixString + operatorsv1.OperatorStatusTypeProgressing
+	handleCondition(operatorConfig, conditionType, prefixString, err)
 }
 
-func (c *consoleOperator) HandleAvailable(operatorConfig *operatorsv1.Console, prefix string, err error) {
-	conditionType := prefix + operatorsv1.OperatorStatusTypeAvailable
-	handleCondition(operatorConfig, conditionType, prefix, err)
+func (c *consoleOperator) HandleAvailable(operatorConfig *operatorsv1.Console, prefix ConditionPrefix, err error) {
+	prefixString := string(prefix)
+	conditionType := prefixString + operatorsv1.OperatorStatusTypeAvailable
+	handleCondition(operatorConfig, conditionType, prefixString, err)
+}
+
+// syncErrors are internal, they imply progressing to us.
+// but any other type of error is a real problem, thus should be
+// degraded.
+// Can we use the "if" here, however?
+// This is likely dangerous.  We prob need a helper func instead.
+func (c *consoleOperator) HandleProgressingOrDegraded(operatorConfig *operatorsv1.Console, prefix ConditionPrefix, err error) {
+	if errors.IsSyncError(err) {
+		c.HandleDegraded(operatorConfig, ConsoleRouteSync, nil)
```

Comment:
Not sure if passing `ConsoleRouteSync` to all the `HandleProgressing` and `HandleDegraded` is right. 
Shouldn't we be passing the `prefix` ?